qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/AppointmentSteps.java

```diff
@@ -0,0 +1,240 @@
+		requestAppointmentPage.addAppointmentRequestNote("I need to see a doctor first");
+	}
+
+	@And("a user clicks save button")
+	public void saveAppointmentRequest() {
+		dashboardPage = requestAppointmentPage.saveRequest();
+	}
+
+	@Then("the system adds appointment request into the appointment requests table")
+	public void systemSavesAppointmentRequest() {
+		assertNotNull(dashboardPage.getAppointmentRequestsList());
+	}
+
+	@When("a user clicks on Schedule Appointment link from the Patient dashboard")
+	public void loadManageAppointmentsPage() {
+		manageAppointmentsPage = (ManageAppointmentsPage) dashboardPage.goToManageAppointments();
+	}
+
+	@And("the system loads Manage Appointments page")
+	public void systemLoadsManageAppointmentsPage() {
+		assertTrue(manageAppointmentsPage.containsText(testPatient.identifier));
+	}
+
+	@And("a user clicks on Cancel this Request icon")
+	public void deleteAppointmentRequest() {
+		manageAppointmentsPage.deleteRequest() ;
+	}
+
+	@Then("the system deletes appointment request from the request table")
+	public void systemDeletesAppointmentRequest() {
+		assertTrue(manageAppointmentsPage.containsText("No appointment requests"));
+	}
+
+	@When("a user clicks Appointment scheduling app from the home page")
+	public void loadAppointmentSchedulingPage() {
+		dashboardPage.goToHomePage();
+		appointmentSchedulingPage = (AppointmentSchedulingPage ) homePage.goToAppointmentScheduling();
+	}
+
+	@Then("the system loads the appointment scheduling page")
+	public void systemLoadsAppointmentSchedulingPage() {
+		assertTrue(textExists("Appointment Scheduling"));
+	}
+
+	@And("a user clicks on Manage Provider Schedules app")
+	public void loadManageProviderSchedulesPage() {
+		manageProviderSchedulesPage = (ManageProviderSchedulesPage) appointmentSchedulingPage.goToManageProviderSchedules();
+	}
+
+	@And("the system loads Manage Provider Schedules page")
+	public void systemLoadsManageProviderSchedulesPage() {
+		assertTrue(textExists("Manage Appointment Blocks"));
+	}
+
+	@And("a user fills in the details of the appointment block")
+	public void captureAppointmentBlockDetails() {
+		manageProviderSchedulesPage.selectLocation("Laboratory");
+		manageProviderSchedulesPage.clickOnNextWeekday();
+		manageProviderSchedulesPage.selectLocationBlock("Laboratory");
+		manageProviderSchedulesPage.enterMinimumTimeValue("06", "30");
+		manageProviderSchedulesPage.clickOnStartTimeButton();
+		manageProviderSchedulesPage.enterMaximumTimeValue("09", "30");
+		manageProviderSchedulesPage.clickOnEndTimeButton();
+		manageProviderSchedulesPage.enterService(FIRST_SERVICE_NAME);
+	}
+
+	@And("a user clicks on the save button")
+	public void saveAppointmentBlockDetails() {
+		manageProviderSchedulesPage.clickOnSave();
+	}
+
+	@Then("the system saves the appointment block")
+	public void systemSavesAppointmentBlockDetails() {
+		assertTrue(manageProviderSchedulesPage.containsText(FIRST_SERVICE_NAME));
```

Comment:
How does this prove that the appointment block has been saved?

Reply:
@dkayiwa  before saving the appointment block, Manage Provider Schedules Page does not contain the service name until a block is saved, then it displays the service that has been blocked for appointment.